src/AWS.Deploy.Orchestration/Utilities/DeployedApplicationQueryer.cs

```diff
@@ -50,6 +51,18 @@ public class DeployedApplicationQueryer : IDeployedApplicationQueryer
         }
 
         public async Task<List<CloudApplication>> GetExistingDeployedApplications()
+        {
+            var existingApplications = new List<CloudApplication>();
+            existingApplications.AddRange(await GetExistingCloudFormationStacks());
+            existingApplications.AddRange(await GetExistingBeanstalkEnvironments());
```

Comment:
We should only be getting beanstalk environments if we are deploying a web application.

Reply:
If the client's application is not a web app, then the new _"Hidden"_ recipe will not be part of the recommendation list. Due to this all existing Beanstalk environments will fail the [IsCompatible check](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.Orchestration/Utilities/DeployedApplicationQueryer.cs#L151) and will not be available for selection.

Reply:
Added an overload for `GetExistingDeployedApplications` that accepts a list of DeploymentTypes and acts as a filter for which resources to query.

Reply:
This version was being used by the `ListDeployments` command. I agree that this can be misused. Got rid of this and updated the `ListDeployments` command to use the new version.